File: Com.Fubei.Api.OpenApi/Net/Interceptor/FubeiOpenApiInterceptor.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using Com.Fubei.Api.OpenApi;
using Com.Illuminati.Galileo.Attr;
using Com.Illuminati.Galileo.Biz.MerchantApi;
using Com.Illuminati.Galileo.Constants;
using Com.Illuminati.Galileo.Exceptions;
using Com.Illuminati.Galileo.Model;
using Com.Illuminati.Galileo.Net.HttpClient;
using Com.Illuminati.Galileo.Utils;
using Newtonsoft.Json;

namespace Com.Illuminati.Galileo.Net.Interceptor
{
    public class FubeiOpenApiInterceptor : AbsFubeiOpenApiInterceptor<OpenApiResponse>
    {
        protected override string GetApiHost()
        {
            return string.IsNullOrEmpty(OpenApiStorage.Instance.OpenApiHost) ? 
                ApiConstants.FubeiOpenapiHost : OpenApiStorage.Instance.OpenApiHost;
        }

        protected override OpenApiResponse DoRequest(string host, ApiAttribute attr, List<KeyValuePair<string, object>> paramList)
        {
            return HttpClientFactory.HttpRequestForOpenApi.PostRequest(host, attr, paramList);
        }

        protected override object AfterExecution(OpenApiResponse data, Type genericType, RequestCarrier carrier)
        {
            if (data.ResultCode == ApiConstants.SuccessCode)
            {
                // 数据解析
                return JsonConvert.DeserializeObject(data.Data.ToString(), genericType);
            }
            else
            {
                //ReflectionUtil.Invoke(returnValue, ApiConstants.Reject,
                //    new object[] { new BizException(data.ResultCode ?? -9999, data.SubCode, data.ResultMessage, data.Data) });
                throw new BizException(data.ResultCode ?? -9999, data.SubCode, data.ResultMessage, data.Data);
            }
        }
    }
}

File: Com.Fubei.OpenApi.Vb/Net/Interceptor/FubeiOpenApiInterceptor.vb
Imports Com.Fubei.Api.OpenApi.Biz
Imports Com.Fubei.Api.OpenApi.Net.HttpClient
Imports Com.Illuminati.Galileo.Attr
Imports Com.Illuminati.Galileo.Constants
Imports Com.Illuminati.Galileo.Exceptions
Imports Com.Illuminati.Galileo.Model
Imports Com.Illuminati.Galileo.Net.Interceptor
Imports Newtonsoft.Json

Namespace Net.Interceptor
    Public Class FubeiOpenApiInterceptor
        Inherits AbsFubeiOpenApiInterceptor(Of OpenApiResponse)

        Protected Overrides Function GetApiHost() As String
            Return If(String.IsNullOrEmpty(OpenApiStorage.Instance.OpenApiHost), ApiConstants.FubeiOpenapiHost, OpenApiStorage.Instance.OpenApiHost)
        End Function

        Protected Overrides Function DoRequest(ByVal host As String, ByVal attr As ApiAttribute, ByVal paramList As List(Of KeyValuePair(Of String, Object))) As OpenApiResponse
            Return HttpClientFactory.HttpRequestForOpenApi.PostRequest(host, attr, paramList)
        End Function

        Protected Overrides Function AfterExecution(ByVal data As OpenApiResponse, ByVal genericType As Type, ByVal carrier As RequestCarrier) As Object
            If data.ResultCode = ApiConstants.SuccessCode Then
                Return JsonConvert.DeserializeObject(data.Data.ToString(), genericType)
            Else
                Throw New BizException(If(data.ResultCode, -9999), data.SubCode, data.ResultMessage, data.Data)
            End If
        End Function
    End Class
End Namespace
